src/storage/posix/storage.c

```diff
@@ -452,28 +463,39 @@ storagePosixPathRemove(THIS_VOID, const String *path, bool recurse, StorageInter
 
     MEM_CONTEXT_TEMP_BEGIN()
     {
-        // Recurse if requested
-        if (recurse)
-        {
-            StoragePosixPathRemoveData data =
-            {
-                .driver = this,
-                .path = path,
-            };
+        bool removed = false;
 
-            // Remove all sub paths/files
-            storageInterfaceInfoListP(this, path, storageInfoLevelExists, storagePosixPathRemoveCallback, &data);
-        }
-
-        // Delete the path
-        if (rmdir(strZ(path)) == -1)
+        do
         {
-            if (errno != ENOENT)                                                                                    // {vm_covered}
-                THROW_SYS_ERROR_FMT(PathRemoveError, STORAGE_ERROR_PATH_REMOVE, strZ(path));                        // {vm_covered}
+            // If path could not be removed
+            if (rmdir(strZ(path)) == -1)
+            {
+                // Path does not exist
+                if (errno == ENOENT)
```

Comment:
In the original implementation, ENOENT would result in returning true.  New implementation is returning false for ENOENT?

Reply:
OK.  Ignore the above.